### PR TITLE
fix: running tests

### PR DESCRIPTION
**Pull request by matthieuRioual**
Writing test for every situation cases

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  'extends': [
+    'google',
+  ],
+  'parserOptions': {
+    'ecmaVersion': 12,
+    'sourceType': 'module',
+  },
+  'rules': {
+  },
+};
```
Comment (fabiendv) [inline .eslintrc.js]: Why did you choose this configuration?

File: .prettierrc.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    
+}
```
Comment (fabiendv) [inline .prettierrc.json]: Do not push empty file

File: test/response-mook1.json [hunk 3]
```diff
@@ -0,0 +1,62 @@
+{
```
Comment (fabiendv) [inline test/response-mook1.json]: Put a real name instead of numbers
eg: response-mook-user-tasks

**Review by fabiendv — CHANGES_REQUESTED** [review]

**Review by fabiendv — CHANGES_REQUESTED** [review]
- [x]  Set up hard and confusing. (API username?) README and questions/help into the configuration need to be updated

- [ ]  The applet should not send signal when no notifications. Send a signal only if smth happened.

- [ ]  All code errors (`cannot read property of undefined`) must be prevent by english langage (eg. `Ouups. Please check the configuration`)

- [x]  I would prefer calling this applet Discourse and saying into the README that only admins can use it. 

- [x] The image should be the same size and border than Asana's applet. (You can find a good example of README as well)

- [x] Maybe explain how to set up admin key (or a link) because it's too hard (with too much config)

- [x]  Update description `Make blinking key...` :> Look at Asana

- [x]  Update displayName :> `Discourse`

File: index.js [hunk 4]
```diff
@@ -1,131 +1,140 @@
...
-    }
-    //if the answer is an error
-    else{
-      return response;
+                signal = new q.Signal({
+                    points: [[new q.Point(warnerColor, warnerEffect)]],
+                    name: 'Discourse',
+                    message: message,
+                    link: {
+                        url:
```
Comment (Mubramaj) [inline index.js]: Could use \`${host}/u/${this.config.username}/notifications?filter=unread\` easier to read

File: index.js [hunk 5]
```diff
@@ -1,131 +1,140 @@
...
-    //generate the q signal
-    const upEffect = this.config.upEffect || "SET_COLOR";
-    const downEffect = this.config.upEffect || "BLINK";
-    const upColor = this.config.upColor || "#00FF00";
-    const downColor = this.config.downColor || "#FF0000";
-    let signal=this.generateSignal(notifications,upColor,downColor,upEffect,downEffect);
-    return signal;
-    
-  }
+    // main function running
```
Comment (Mubramaj) [inline index.js]: This is not the right format for a js method comment

File: package.json [hunk 6]
```diff
@@ -1,114 +1,100 @@
...
-      }
-
-    ]
-  },
-  "devDependencies": {
-    "mocha": "^5.2.0"
-  }
+    "dependencies": {
+        "daskeyboard-applet": "^2.11.3",
+        "dotenv": "^8.2.0",
```
Comment (Mubramaj) [inline package.json]: Why do you need dotenv ?

File: test/response-mook-1-notif.json [hunk 7]
```diff
@@ -0,0 +1,62 @@
+{
+    "notifications": [
```
Comment (Mubramaj) [inline test/response-mook-1-notif.json]: Name of file should be more explicit `response-mook-1` not working.

Example of good name -> goodResponseWith2NotificationsMock

**Review by Mubramaj — CHANGES_REQUESTED** [review]

**Comment by fabiendv** [comment]
- [x] Not working without entering http
- [x] Removed API Key usename (unused)
- [x] mysuperusername becomes johnDoe

**Comment by fabiendv** [comment]
- [x] fix: click `show in Discourse` go to blank link

**Comment by fabiendv** [comment]
- [x] dotenv has been removed. But not into `yarn.lock`. Please delete the file and run yarn again.
- [x] icon size should be 512x512
- [x] image should be updated (same layout than Asana's)

**Comment by fabiendv** [comment]
- [x] When adding API key. The link goes to github : update to `<your-forum-url>.com/admin`. I admit the solution is not good and ugly, but a github link is worst because the repo can be deleted and the user won't have an easy help.

**Comment by fabiendv** [comment]
Readme:

- [x] title is not standard. Please check Asana.
- [ ] should respect MDlint
- [x] first paragraph is even to readme_end_user

File: package.json [hunk 8]
```diff
@@ -43,7 +43,7 @@
         "authorization": {
             "type": "apiKey",
             "hint": "You need admin access to generate an API key",
-            "supportUrl": "https://github.com/discourse/discourse/blob/master/docs/ADMIN-QUICK-START-GUIDE.md"
+            "supportUrl": "<your-forum-url>.com/admin"
```
Comment (fabiendv) [inline package.json]: Add https.
+ Add go to <your-forum-url> into the hint

**Review by fabiendv — CHANGES_REQUESTED** [review]

**Comment by fabiendv** [comment]
- [x] Update tests to make them pass
- [x] Prevent `notifications of undefined` error when user enter a bad forum url.
- [x] Popup link is not working : Maybe you can add `https://` before `${host}

**Pull request merged**